src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -813,24 +814,34 @@ void GLHEBase::simulate(EnergyPlusData &state,
 
 //******************************************************************************
 
-PlantComponent *GLHEBase::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
+GLHEBase *GLHEBase::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
 {
     if (state.dataGroundHeatExchanger->GetInput) {
         GetGroundHeatExchangerInput(state);
         state.dataGroundHeatExchanger->GetInput = false;
     }
     if (objectType == DataPlant::PlantEquipmentType::GrndHtExchgSystem) {
-        for (auto &ghx : state.dataGroundHeatExchanger->verticalGLHE) {
-            if (ghx.name == objectName) {
-                return &ghx;
-            }
-        }
```

Comment:
[src/EnergyPlus/GroundHeatExchangers.cc:824]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.